code/obj/item/mob_parts/artifact_parts.dm

```diff
@@ -0,0 +1,528 @@
+
+	left
+		name = "martian left arm"
+		slot = "l_arm"
+		side = "left"
+		icon_state = "arm-martian-L"
+		handlistPart = "arm-martian-L-attached"
+
+	right
+		name = "martian right arm"
+		slot = "r_arm"
+		side = "right"
+		icon_state = "arm-martian-R"
+		handlistPart = "arm-martian-R-attached"
+
+/obj/item/parts/artifact_parts/leg/martian
+	artifact_type = "martian"
+
+	left
+		name = "martian left leg"
+		slot = "l_leg"
+		side = "left"
+		step_image_state = "footprintsL"
+		icon_state = "leg-martian-L"
+		handlistPart = "leg-martian-L-attached"
+		movement_modifier = /datum/movement_modifier/martian_legs/left
+
+	right
+		name = "martian right leg"
+		slot = "r_leg"
+		side = "right"
+		step_image_state = "footprintsR"
+		icon_state = "leg-martian-R"
+		handlistPart = "leg-martian-R-attached"
+		movement_modifier = /datum/movement_modifier/martian_legs/right
+
+/obj/item/parts/artifact_parts/arm/precursor
+	artifact_type = "precursor"
+
+	on_attach()
+		if (!..())
+			return
+		var/mob/living/carbon/human/H = holder
+		if (istype(H.limbs.get_limb("l_arm"), /obj/item/parts/artifact_parts/arm/precursor/left) && istype(H.limbs.get_limb("r_arm"), /obj/item/parts/artifact_parts/arm/precursor/right))
+			src.holder.addAbility(/datum/targetable/artifact_limb_ability/precursor_heal)
+
+	on_remove()
+		if (!..())
+			return
+		src.holder.removeAbility(/datum/targetable/artifact_limb_ability/precursor_heal)
+
+	left
+		name = "precursor left arm"
+		slot = "l_arm"
+		side = "left"
+		icon_state = "arm-precursor-L"
+		handlistPart = "arm-precursor-L-attached"
+
+	right
+		name = "precursor right arm"
+		slot = "r_arm"
+		side = "right"
+		icon_state = "arm-precursor-R"
+		handlistPart = "arm-precursor-R-attached"
+
+/obj/item/parts/artifact_parts/leg/precursor
+	artifact_type = "precursor"
+
+	on_attach()
+		if (!..())
+			return
+		if (src.side == "left")
+			RegisterSignal(src.holder, COMSIG_MOVABLE_MOVED, .proc/precursor_move_L)
+		else
+			RegisterSignal(src.holder, COMSIG_MOVABLE_MOVED, .proc/precursor_move_R)
+
+	on_remove()
+		if (!..())
+			return
+		UnregisterSignal(src.holder, COMSIG_MOVABLE_MOVED)
+
+	proc/precursor_move_L(mob/living/carbon/human/H, oldLoc, direction)
+		if (H.lying)
+			return
+
+		var/turf/T = get_turf(H)
+
+		var/image/I = image('icons/obj/artifacts/artifactLimbs.dmi', icon_state = "footprints-L", layer = DECAL_LAYER, dir = H.dir)
+		T.UpdateOverlays(I, "precursorfootprints-L-[ref(I)]")
+		SPAWN(1 SECOND)
+			T.UpdateOverlays(null, "precursorfootprints-L-[ref(I)]")
+
+	proc/precursor_move_R(mob/living/carbon/human/H, oldLoc, direction)
+		if (H.lying)
+			return
+
+		var/turf/T = get_turf(H)
+
+		var/image/I = image('icons/obj/artifacts/artifactLimbs.dmi', icon_state = "footprints-R", layer = DECAL_LAYER, dir = H.dir)
+		T.UpdateOverlays(I, "precursorfootprints-R-[ref(I)]")
+		SPAWN(1 SECOND)
+			T.UpdateOverlays(null, "precursorfootprints-R-[ref(I)]")
```

Comment:
Wonder if these would look better as a decal in general that cleans itself after a short period of time?  Additionally I'd rather they slowly appear as they appear before the sprite glides over the tile in question.

Reply:
The reason I didn't go with decals was to prevent several decals from appearing on a tile from right click, even though it might not happen much, depending on time they disappear after. Idk, I feel like fading in might seem a bit odd considering they could fade in after you've passed the tile. I was thinking of making them have a glow effect, but only problem was it was possibly a lot of sprites, if a problem. Maybe fading out?

Reply:
Alternative place them on the old location? That way they don't appear before the sprite even glides to the tile.

Reply:
Sure

Reply:
Yeah I can do that, though should say that blood and paint tracking rounds already give you prints that work like how this does right now